manager/orchestrator/services.go

```diff
@@ -81,17 +81,37 @@ func (r *ReplicatedOrchestrator) reconcile(ctx context.Context, service *api.Ser
 		return
 	}
 
+	// indicator of service mode change
+	modeChanged := false
 	runningTasks := make([]*api.Task, 0, len(tasks))
 	runningInstances := make(map[uint64]struct{}) // this could be a bitfield...
 	for _, t := range tasks {
 		// Technically the check below could just be
 		// t.DesiredState <= api.TaskStateRunning, but ignoring tasks
 		// with DesiredState == NEW simplifies the drainer unit tests.
 		if t.DesiredState > api.TaskStateNew && t.DesiredState <= api.TaskStateRunning {
+			if t.Slot == 0 {
+				modeChanged = true
+				break
+			}
 			runningTasks = append(runningTasks, t)
 			runningInstances[t.Slot] = struct{}{}
 		}
 	}
+
+	// if service mode is changed, clean up existing tasks because slots need to renumber
+	if modeChanged {
+		_, err = r.store.Batch(func(batch *store.Batch) error {
+			r.removeTasks(ctx, batch, service, tasks)
```

Comment:
I think updating mode is a rare event and not worth optimization. (As we discuss, current update function doesn't update `Slot` number.)